processing/src/main/java/org/apache/druid/query/rowsandcols/column/LongArrayColumn.java

```diff
@@ -0,0 +1,201 @@
+      return (T) (ColumnValueSwapper) (lhs, rhs) -> {
+        long tmp = vals[lhs];
+        vals[lhs] = vals[rhs];
+        vals[rhs] = tmp;
+      };
+    }
+    return null;
+  }
+
+  private class MyColumnAccessor implements BinarySearchableAccessor
+  {
+    @Override
+    public ColumnType getType()
+    {
+      return ColumnType.LONG;
+    }
+
+    @Override
+    public int numRows()
+    {
+      return vals.length;
+    }
+
+    @Override
+    public boolean isNull(int rowNum)
+    {
+      return false;
+    }
+
+    @Override
+    public Object getObject(int rowNum)
+    {
+      return vals[rowNum];
+    }
+
+    @Override
+    public double getDouble(int rowNum)
+    {
+      return vals[rowNum];
+    }
+
+    @Override
+    public float getFloat(int rowNum)
+    {
+      return vals[rowNum];
+    }
+
+    @Override
+    public long getLong(int rowNum)
+    {
+      return vals[rowNum];
+    }
+
+    @Override
+    public int getInt(int rowNum)
+    {
+      return (int) vals[rowNum];
+    }
+
+    @Override
+    public int compareRows(int lhsRowNum, int rhsRowNum)
+    {
+      return Long.compare(vals[lhsRowNum], vals[rhsRowNum]);
+    }
+
+
+    @Override
+    public FindResult findNull(int startIndex, int endIndex)
+    {
+      return FindResult.notFound(endIndex);
+    }
+
+    @Override
+    public FindResult findDouble(int startIndex, int endIndex, double val)
+    {
+      return findLong(startIndex, endIndex, (int) val);
+    }
+
+    @Override
+    public FindResult findFloat(int startIndex, int endIndex, float val)
+    {
+      return findLong(startIndex, endIndex, (int) val);
+    }
+
+    @Override
+    public FindResult findLong(int startIndex, int endIndex, long val)
+    {
+      if (vals[startIndex] == val) {
+        int end = startIndex + 1;
+
+        while (end < endIndex && vals[end] == val) {
+          ++end;
+        }
+        return FindResult.found(startIndex, end);
+      }
+
+      int i = Arrays.binarySearch(vals, startIndex, endIndex, val);
+      if (i > 0) {
+        int foundStart = i;
+        int foundEnd = i + 1;
+
+        while (foundStart - 1 >= startIndex && vals[foundStart - 1] == val) {
+          --foundStart;
+        }
+
+        while (foundEnd < endIndex && vals[foundEnd] == val) {
+          ++foundEnd;
+        }
+
+        return FindResult.found(foundStart, foundEnd);
+      } else {
+        return FindResult.notFound(-(i + 1));
+      }
+    }
+
+    public FindResult findInt(int startIndex, int endIndex, int val)
+    {
+      return findLong(startIndex, endIndex, val);
+    }
+
+    @Override
+    public FindResult findString(int startIndex, int endIndex, String val)
+    {
+      return findLong(startIndex, endIndex, (int) Numbers.tryParseLong(val, 0));
+    }
+
+    @Override
+    public FindResult findComplex(int startIndex, int endIndex, Object val)
+    {
+      return findLong(startIndex, endIndex, Numbers.tryParseLong(val, 0));
```

Comment:
i know this isn't really new, but why doesn't this need to distinguish between the case where `val` is parseable into a number and not?

Reply:
The meta answer to your question is because it's not an interface that's used for anything actually meaningful yet.

The actual answer to your question is that the one place that is trying to use this is first checking for `null` and then calling `findNull`  It then does a switch on the searcher's type and calls the relevant method for that.  The type of the `LongArrayColumn` is `LONG` so it won't actually call `findComplex()` and will call `findLong()` instead.  This is fundamentally putting the responsibility on the call site to "do the right thing".  Is that the correct behavior or not?  I dunno, but it's an experimental interface that you would have to be incredibly intrepid to actually hit, so it's a detail that perhaps doesn't matter.

If your suggestion is to switch this to a `NotYetImplemented` exception, I'd probably agree with that suggestion.

Reply:
Changed the numberColumns to throw an exception instead of using a 0 if the string is unparsable for `findComplex` and `findString`